### ISSUE TITLE
Results for response selector not consistent across runs

### ISSUE THREAD

**Issue by dakshvar22** [RasaHQ/rasa#6380]
**Rasa version**: master

**Issue**: [Metabase dashboard](https://metabase.rasa.com/dashboard/166) shows that the results for response selector have varied across multiple runs for some configurations. This should ideally not have happened since we use a random seed always in the configurations.

**Comment by dakshvar22** [comment]
Update: It looks like performance for intent classification and entity recognition have also varied by a very small margin in the [last few runs](https://metabase.rasa.com/dashboard/166)

**Comment by dakshvar22** [comment]
Update: I am not able to reproduce difference in numbers on different runs locally on a MacOS. As a test case, I took the configuration `Sparse + DIET(bow) + ResponseSelector(bow)` and Carbon Bot dataset and trained 4 models with the same random seed. All 4 models when evaluated on the test data had the exact same numbers on all metrics.

However, when I run the same test case as a model regression test, the results differ: [run1](https://github.com/RasaHQ/rasa/pull/6850#issuecomment-701373461) and [run2](https://github.com/RasaHQ/rasa/pull/6850#issuecomment-701394181). The only thing that is differs is the OS on which the tests are being run. Locally it's MacOS and for model regression tests it's Linux. So, the current hypothesis is that the random seed setting is not working well on Linux platforms.

**Comment by dakshvar22** [comment]
It looks like running operations on GPU are not guaranteed to produce exactly the same output on different runs as the [Keras documentation](https://keras.io/getting_started/faq/#how-can-i-obtain-reproducible-results-using-keras-during-development) says.
Validated this with multiple runs([1](https://github.com/RasaHQ/rasa/pull/6850#issuecomment-701486503), [2](https://github.com/RasaHQ/rasa/pull/6850#issuecomment-701496169), [3](https://github.com/RasaHQ/rasa/pull/6850#issuecomment-701506919) of model regression tests with a CPU runner. This validates that there is nothing wrong with the code and some deviation is always expected when training and evaluating on a GPU.
